mrmustard/lab_dev/circuit_components.py

```diff
@@ -488,11 +488,13 @@ def fock(self, shape: int | Sequence[int] | None = None, batched=False) -> Compl
                     f"Expected Fock shape of length {num_vars}, got length {len(shape)}"
                 )
             if self.representation.ansatz.polynomial_shape[0] == 0:
-                arrays = [math.hermite_renormalized(A, b, c, shape) for A, b, c in zip(As, bs, cs)]
+                arrays = [
+                    math.hermite_renormalized(A, b, c, shape=shape) for A, b, c in zip(As, bs, cs)
```

Comment:
`shape` in `math.hermite_renormalized` should be passed as a keyword argument to prevent `tf.custom_gradient` from capturing it as a trainable argument.